pkg/detectors/uri/uri_test.go

```diff
@@ -30,6 +31,11 @@ func TestURI_Pattern(t *testing.T) {
 			input: fmt.Sprintf("%s token = '%s'", keyword, validPattern),
 			want:  []string{validPattern},
 		},
+		{
+			name:  "valid pattern - capture two outputs",
+			input: fmt.Sprintf("%s token = '%s'", keyword, validPattern2),
+			want:  []string{"http://username:password@127.0.0.1", "http://username:password@127.0.0.1"},
```

Comment:
While not related to #3815, the results should be deduplicated.

```suggestion
			want:  []string{"http://username:password@127.0.0.1"},
```

Reply:
Done